app/views/users/edit.html.erb

```diff
@@ -31,3 +31,7 @@
                email_addresses: current_user.email_addresses } %>
   </div>
 </div>
+
+<% if controller.request.env["HTTP_REFERER"].present? %>
+  <%= link_to 'Back', :back %>
```

Comment:
actually we probably need to store this in the session?  from what i remember, the back link still worked after the profile form is saved...